### PR TITLE
DOCSP-43190 -- Add Automation example for local Atlas instance in Docker

### PR DESCRIPTION
**Pull request by jvincent-mongodb**
Add Automation example for local Atlas instance in Docker.

- [DOCSP-number](https://jira.mongodb.org/browse/DOCSP-43190)
- [STAGING](https://preview-mongodbjvincentmongodb.gatsbyjs.io/atlas-cli/DOCSP-43190/atlas-cli-deploy-docker/#run-the-docker-image.) Under the tab "Automate Connection"

- [LATEST BUILD LOG](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=66e339731b6f794a2e79a6f9)

### Self-Review Checklist

- [ ] Check that the submodule pulled in the right changes (if applicable).
- [ ] [Define](https://wiki.corp.mongodb.com/display/DE/Taxonomy+tagging+instructions) taxonomy [values](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) at top of page.
- [ ] Add genre facets (tutorial or reference), as in this [example PR](https://github.com/10gen/cloud-docs/pull/5042).
- [ ] Add programmingLanguage (if necessary).
- [ ] Add meta keywords (if necessary).
- [ ] Resolve any new warnings or errors in the build.
- [ ] Proofread for spelling and grammatical errors.
- [ ] Check staging for rendering issues.
- [ ] Confirm links are working.

### CODE REVIEW

**Review by jeroenvervaeke — APPROVED** [review]

File: source/atlas-cli-deploy-docker.txt [hunk 1]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
+         .. tab:: Automate Connection
+            :tabid: automate-connection
+
+            To automate a containerized deployment of Atlas, you'll need to wait 
+            for the container to be in a healthy state before you can connect to Atlas. 
+
+            The following example demonstrates deploying an Atlas image in Docker
+            and polling Docker to check the state of the container. Once the 
+            container is in a healthy state, the script atuomates a connection 
+            to your Atlas instance with Mongosh.
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:80]: Can we add a link here for Mongosh?

File: source/atlas-cli-deploy-docker.txt [hunk 2]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
-      The logs display as the Docker image runs.
+         .. tab:: Automate Connection
+            :tabid: automate-connection
+
+            To automate a containerized deployment of Atlas, you'll need to wait 
+            for the container to be in a healthy state before you can connect to Atlas. 
+
+            The following example demonstrates deploying an Atlas image in Docker
+            and polling Docker to check the state of the container. Once the 
+            container is in a healthy state, the script atuomates a connection
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:79]: automates -- typo

File: source/atlas-cli-deploy-docker.txt [hunk 3]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
+            and polling Docker to check the state of the container. Once the 
+            container is in a healthy state, the script atuomates a connection 
+            to your Atlas instance with Mongosh. 
+
+            a. Create a file called ``mongodb-atlas-local.sh``, and paste the 
+               following script into your new file. 
+
+               .. code-block:: sh
+
+                  # start mongodb-atlas-local container
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:87]: Nit: do we want to start all comments with Init caps?

File: source/atlas-cli-deploy-docker.txt [hunk 4]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
 
             .. code-block:: sh
 
                docker run -e MONGODB_INITDB_ROOT_USERNAME=user -e MONGODB_INITDB_ROOT_PASSWORD=pass -p 27017:27017 mongodb/mongodb-atlas-local
 
-      The logs display as the Docker image runs.
+         .. tab:: Automate Connection
+            :tabid: automate-connection
+
+            To automate a containerized deployment of Atlas, you'll need to wait
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:74]: Does this repo use |service| for Atlas? If yes, please use it here.

File: source/atlas-cli-deploy-docker.txt [hunk 5]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
             .. code-block:: sh
 
                docker run -e MONGODB_INITDB_ROOT_USERNAME=user -e MONGODB_INITDB_ROOT_PASSWORD=pass -p 27017:27017 mongodb/mongodb-atlas-local
 
-      The logs display as the Docker image runs.
+         .. tab:: Automate Connection
+            :tabid: automate-connection
+
+            To automate a containerized deployment of Atlas, you'll need to wait 
+            for the container to be in a healthy state before you can connect to Atlas.
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:75]: Same in this line and below on lines 77 and 80

File: source/atlas-cli-deploy-docker.txt [hunk 6]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
+                  wait "$CONTAINER_ID"
+                  EXPOSED_PORT=$(docker inspect --format='{{ (index (index .NetworkSettings.Ports "27017/tcp") 0).HostPort }}' "$CONTAINER_ID")
+                  
+                  # build the connectionstring
+                  CONNECTION_STRING="mongodb://127.0.0.1:$EXPOSED_PORT/test?directConnection=true"
+                  
+                  # example usage of the connection string to connect to mongosh
+                  mongosh "$CONNECTION_STRING"
+
+            b. Run the following command to make the file executable.
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:129]: Suggestion: Run the following command to change permissions and make the script executable.

File: source/atlas-cli-deploy-docker.txt [hunk 7]
```diff
@@ -54,21 +54,89 @@ Create a Local Atlas Deployment with Docker
...
+                  # example usage of the connection string to connect to mongosh
+                  mongosh "$CONNECTION_STRING"
+
+            b. Run the following command to make the file executable. 
+
+               .. code-block:: sh
+
+                  chmod +x mongodb-atlas-local.sh
+
+            c. Run the executable.
```
Comment (JuliaMongo) [inline source/atlas-cli-deploy-docker.txt:135]: Run the script.

**Review by JuliaMongo — APPROVED** [review]
LGTM with a few nits. Thank you!

**Pull request merged**